main/src/main/java/org/mobilitydata/gtfsvalidator/validator/DuplicateFareRuleZoneIdFieldsValidator.java

```diff
@@ -41,27 +42,36 @@
 
   @Override
   public void validate(NoticeContainer noticeContainer) {
-    final Map<String, GtfsFareRule> fareRuleByZoneIdFieldsCombination =
+    final Map<Integer, GtfsFareRule> fareRuleByZoneIdFieldsCombination =
         new HashMap<>(fareRuleTable.entityCount());
     fareRuleTable
         .getEntities()
         .forEach(
             fareRule -> {
-              String fieldsCombination =
-                  fareRule.routeId()
-                      + fareRule.originId()
-                      + fareRule.containsId()
-                      + fareRule.destinationId();
-              if (fareRuleByZoneIdFieldsCombination.containsKey(fieldsCombination)) {
-                noticeContainer.addValidationNotice(
-                    new DuplicateFareRuleZoneIdFieldsNotice(
-                        fareRule.csvRowNumber(),
-                        fareRule.fareId(),
-                        fareRuleByZoneIdFieldsCombination.get(fieldsCombination).csvRowNumber(),
-                        fareRuleByZoneIdFieldsCombination.get(fieldsCombination).fareId()));
-              } else {
-                fareRuleByZoneIdFieldsCombination.put(fieldsCombination, fareRule);
+              int hash = getHash(fareRule);
+              GtfsFareRule otherFareRule = fareRuleByZoneIdFieldsCombination.get(hash);
+              if (otherFareRule == null) {
+                fareRuleByZoneIdFieldsCombination.put(hash, fareRule);
+                return;
               }
```

Comment:
Use putIfAbsent instead of combination of get + put. This way you make the code shorter and avoid the temporary `hash` variable.

Reply:
Fixed in https://github.com/MobilityData/gtfs-validator/pull/811/commits/8b867b4dc69652fc75a9065dbb1c9f7e7af3809f

Reply:
@aababilov Just curious - in https://github.com/MobilityData/gtfs-validator/blob/3ceadcb1ee9626499de40e68c6cc88aa034688df/core/src/main/java/org/mobilitydata/gtfsvalidator/parsing/FieldCache.java#L60 ...you mentioned that:

>    // Benchmarks show that computeIfAbsent() is about 20% more expensive than calling get() and
    // put().

Do you know if the same applies to `putIfAbsent()`?

This post seems to think so, although the video it references is from 2009 - https://www.javaer101.com/en/article/4544460.html.